internal/mempool/mempool.go

```diff
@@ -839,6 +839,9 @@ func (txmp *TxMempool) removeTx(wtx *WrappedTx, removeFromCache bool) {
 
 	if removeFromCache {
 		txmp.cache.Remove(wtx.tx)
+		if wtx.expiredCallback != nil {
```

Comment:
I think this logic should be outside `if removeFromCache {}`. The `cache` here is an unfortunate naming, but it's basically just for deduplication. So if `removeFromCache` is configured false, it means the node doesn't allow a tx to be re-submitted no matter what, which i believe is the case for some validators/nodes (for perf reason). In that case, we still want to invoke the expired callback, because otherwise the caller would not be able to send  a tx (even with a different payload and thus should not be dedup-ed) using the same nonce of the expired tx